tools/install-powershell.ps1

```diff
@@ -0,0 +1,114 @@
+<#
+.Synopsis
+    Install PowerShell Core on Windows.
+.DESCRIPTION
+    By default, the latest PowerShell Core release package will be installed.
+    If '-Daily' is specified, then the latest PowerShell Core daily package will be installed.
+.Parameter Destination
+    The destination path to install PowerShell Core to.
+.Parameter Daily
+    Install PowerShell Core from the daily build.
+    Note that the 'PackageManagement' module is required to install a daily package.
+.Parameter NotOverwrite
+    Do not overwrite the destination folder if it already exists.
+.Parameter AddToPath
+    Add the absolute destination path to the 'User' scope environment variable 'Path'
+#>
+[CmdletBinding()]
+param(
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $Destination,
+
+    [Parameter()]
+    [switch] $Daily,
+
+    [Parameter()]
+    [switch] $NotOverwrite,
+
+    [Parameter()]
+    [switch] $AddToPath
+)
+
+Set-StrictMode -Version latest
+$ErrorActionPreference = "Stop"
+
+if (-not $Destination) {
+    $Destination = "~/pscore"
+    if ($Daily) {
+        $Destination = "${Destination}-daily"
+    }
+}
+
+if (Test-Path -Path $Destination) {
+    if ($NotOverwrite) {
+        throw "Destination folder '$Destination' already exist. Use a different path or omit '-NotOverwrite' to overwrite."
+    }
+    Remove-Item -Path $Destination -Recurse -Force
+}
+New-Item -ItemType Directory -Path $Destination -Force > $null
+$Destination = Resolve-Path -Path $Destination | ForEach-Object -MemberName Path
+Write-Verbose "Destination: $Destination" -Verbose
+
+$architecture = Get-CimInstance win32_operatingsystem | ForEach-Object {
```

Comment:
This seems overkill to check architecture, I would simply use:

```powershell
switch ($env:PROCESSOR_ARCHITECTURE) {
   ...
}

Reply:
`$env:PROCESSOR_ARCHITECTURE` shows `x86` from a x86 windows powershell on a x64 machine. That's why I turned to `win32_operatingsystem`.

Reply:
It seems like someone deliberately running x86 powershell would expect to install x86 pscore.  Not a blocking issue, but seems unnecessary.

Reply:
I will change back to `$env:PROCESSOR_ARCHITECTURE`. It would be rare for someone to run a x86 powershell on a x64 machine anyways. We can fix it If it turns out to be an issue.